src/dymaptic.GeoBlazor.Core/Model/OAuthAuthentication.cs

```diff
@@ -0,0 +1,99 @@
+using Microsoft.Extensions.Configuration;
+using Microsoft.JSInterop;
+
+
+namespace dymaptic.GeoBlazor.Core.Model;
+
+/// <summary>
+///     A simple class for handling OAuth authentication.
+/// </summary>
+public class OAuthAuthentication : LogicComponent
+{
+    /// <summary>
+    ///     Default Constructor
+    /// </summary>
+    /// <param name="jsRuntime">
+    ///     Injected JavaScript Runtime reference
+    /// </param>
+    /// <param name="config">
+    ///     Injected configuration object
+    /// </param>
+    public OAuthAuthentication(IJSRuntime jsRuntime, IConfiguration config) : base(jsRuntime, config)
+    {
+        _config = config;
+#pragma warning disable CS4014
+        if (!string.IsNullOrEmpty(AppId))
+        {
+            Initialize();
```

Comment:
Yeah, I'm open to alternatives. Basically, I need that Initialize code (the one in the typescript file, really) to run as early as possible. The Esri OAuth code automatically checks the URL for query parameters that are the result of finishing the OAuth workflow, so this needs to run as soon as possible so that those can be processed and the user logged in correctly.

I am firing this here to try to get that to happen. That works about 98% of the time in my testing, so to make it 100% (although that last 2% has a delay) I placed calls to Initialize everywhere, to ensure that we really are ready to go.

Another complication I introduced is trying to follow the existing APIKey pattern in which it could be changed after the Initialization of the app. This meant that I needed to be able to re-call Initialize at any point with a new AppId.

Reply:
@morehavoc did you ever get to explore what we discussed about adding the initialization to aspnetcore middleware?

Reply:
@morehavoc I think we did discuss this further in person, but I am on board with the plan above, with the caveat that we discussed that it would be best to abstract away all differences in setup between Server and WASM. Let me know if you need help implementing this.